File: ProyectoSuscripciones/src/main/java/Models/ModelUser.java
package Models;

import Models.Entities.User;
import Utilities.Util;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import org.apache.http.client.fluent.Form;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;

import javax.swing.*;
import java.util.ArrayList;

import static Models.Entities.User.*;

public class ModelUser {
    String ruta = USERNAME;
    public ModelUser() {

    }

    public User login(String usr, String pass) throws Exception {

        User u = null;

        Form form = Form.form();
        form.add("usr", usr);
        form.add("pass", pass);

        try {
            JSONObject json = Util.requestJsonObj(form, "ModelUser/endPointLogin.php");
            if(json != null && !json.toString().equals("Error")) {

                u = new Gson().fromJson(json.toString(), User.class);

            }

        } catch (Exception e) {
            e.printStackTrace();
        }

        return u;
    }

    public User register(User user) throws Exception {
        String name = user.getName();
        String username = user.getUsername();
        String phone = user.getPhone();
        Integer user_type = user.getUser_type();
        Integer idSucursal = user.getSucursal();
        String email = user.getEmail();
        boolean status = user.isStatus();
        String password = user.getPassword();

        User u = null;


        Form form = Form.form();
        form.add("username", username);
        form.add("name", name);
        form.add("phone", phone);
        form.add("user_type", user_type.toString());
        form.add("Sucursal_idSucursal", idSucursal.toString());
        form.add("email", email);
        form.add("status", status ? "1" : "0");
        form.add("password", password);

        try {
            JSONObject json = Util.requestJsonObj(form, "ModelUser/endPointRegister.php");
            if(json != null && !json.toString().equals("Error")) {
                u = new Gson().fromJson(json.toString(), User.class);
            }
        } catch (Exception e) {
            e.printStackTrace();
            JOptionPane.showMessageDialog(null, "Error");
        }

        return u;
    }

    public void registerAdmin(User user) throws Exception {
        String name = user.getName();
        String username = user.getUsername();
        String phone = user.getPhone();
        Integer user_type = user.getUser_type();
        String email = user.getEmail();
        boolean status = user.isStatus();
        String password = user.getPassword();

        User u = null;


        Form form = Form.form();
        form.add("username", username);
        form.add("name", name);
        form.add("phone", phone);
        form.add("user_type", user_type.toString());
        form.add("Sucursal_idSucursal", null);
        form.add("email", email);
        form.add("status", status ? "1" : "0");
        form.add("password", password);

        try {
            JSONObject json = Util.requestJsonObj(form, "ModelUser/endPointRegister.php");
            if(json != null && !json.toString().equals("Error")) {
                u = new Gson().fromJson(json.toString(), User.class);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public User byId(Integer idUser) throws Exception {
        User u = null;

        Form form = Form.form();
        form.add("idUser", idUser.toString());

        try {
            JSONObject json = Util.requestJsonObj(form, "ModelUser/endPointByIdUser.php");
            if(json != null && !json.toString().equals("Error")) {
                u = new Gson().fromJson(json.toString(), User.class);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        return u;
    }

    public void update(User user) throws Exception {
        Integer idUser = user.getIdUser();
        String name = user.getName();
        String username = user.getUsername();
        String phone = user.getPhone();
        Integer user_type = user.getUser_type();
        Integer idSucursal = user.getSucursal();
        String email = user.getEmail();
        boolean status = user.isStatus();
        String password = user.getPassword();


        Form form = Form.form();
        form.add("idUser", idUser.toString());
        form.add("name", name);
        form.add("username", username);
        form.add("phone", phone);
        form.add("status", status ? "1" : "0");
        form.add("user_type", user_type.toString());
        form.add("Sucursal_idSucursal", idSucursal.toString());
        form.add("email", email);
        form.add("password", password);

        try{
            Util.request(form, "ModelUser/endPointUpdateUser.php");
        } catch (Exception e){
            e.printStackTrace();
        }
    }

    public ArrayList<User> getAll() throws Exception {
        ArrayList<User> users = new ArrayList<User>();

        Form form = Form.form();
        JSONArray array = Util.requestArray(form, "ModelUser/endPointGetAllUser.php");

        if(array != null) {
            java.lang.reflect.Type listType = new TypeToken<ArrayList<User>>() {}.getType();
            users = new Gson().fromJson(array.toString(), listType);
        }

        return users;
    }

    public ArrayList<User> getSellers() throws Exception {
        ArrayList<User> users = new ArrayList<>();

        Form form = Form.form();
        form.add("user_type", "3");

        JSONArray array = Util.requestArray(form, "ModelUser/endPointGetByUserType.php");

        if(array != null) {
            java.lang.reflect.Type listType = new TypeToken<ArrayList<User>>() {}.getType();
            users = new Gson().fromJson(array.toString(), listType);
        }

        return users;
    }

    public ArrayList<User> getAdmins() throws Exception {
        ArrayList<User> users = new ArrayList<>();

        Form form = Form.form();
        form.add("user_type", "1");

        JSONArray array = Util.requestArray(form, "ModelUser/endPointGetByUserType.php");

        if(array != null) {
            java.lang.reflect.Type listType = new TypeToken<ArrayList<User>>() {}.getType();
            users = new Gson().fromJson(array.toString(), listType);
        }

        return users;
    }

    public ArrayList<User> getClients() throws Exception {
        ArrayList<User> users = new ArrayList<>();

        Form form = Form.form();
        form.add("user_type", "2");

        JSONArray array = Util.requestArray(form, "ModelUser/endPointGetByUserType.php");

        if(array != null) {
            java.lang.reflect.Type listType = new TypeToken<ArrayList<User>>() {}.getType();
            users = new Gson().fromJson(array.toString(), listType);
        }

        return users;
    }

    public boolean delete(Integer idUser) throws Exception {
        boolean result = false;
        Form form = Form.form();
        form.add("idUser", idUser.toString());

        try {
            JSONObject json = Util.requestJsonObj(form, "ModelUser/endPointDelete.php");
            if(json != null && !json.toString().equals("Error")) {
                result = new Gson().fromJson(json.toString(), boolean.class);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return result;
    }


}

File: PHP/Models/ModelUser/ModelUser.php
<?php

require 'Models/DBManager.php';

class ModelUser {
    private $db;

    public function __construct() {
        $this->db = new DBManager();
    }

    public function login($usr, $pass) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "SELECT * FROM user WHERE username = ? AND password = SHA1(?) LIMIT 1";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param($stmt, "ss", $usr, $pass);

        mysqli_stmt_execute($stmt);
        $result = mysqli_stmt_get_result($stmt);

        $user = null;
        if ($row = mysqli_fetch_assoc($result)) {
            $user = $row;
        }

        mysqli_stmt_close($stmt);
        $this->db->close($link);

        return $user;
    }

    public function register($user) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "INSERT INTO user (name, username, phone, user_type, Sucursal_idSucursal, email, status, password) 
                VALUES (?, ?, ?, ?, ?, ?, ?, SHA1(?))";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param(
            $stmt, 
            "sssiiiss", 
            $user['name'], 
            $user['username'], 
            $user['phone'], 
            $user['user_type'], 
            $user['Sucursal_idSucursal'], 
            $user['email'], 
            $user['status'], 
            $user['password']
        );

        $success = mysqli_stmt_execute($stmt);
        mysqli_stmt_close($stmt);
        $this->db->close($link);

        if ($success) {
            return $this->login($user['username'], $user['password']);
        }

        return null;
    }

    public function byId($idUser) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "SELECT * FROM user WHERE idUser = ?";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param($stmt, "i", $idUser);

        mysqli_stmt_execute($stmt);
        $result = mysqli_stmt_get_result($stmt);

        $user = null;
        if ($row = mysqli_fetch_assoc($result)) {
            $user = $row;
        }

        mysqli_stmt_close($stmt);
        $this->db->close($link);

        return $user;
    }

    public function update($user) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "UPDATE user SET name = ?, username = ?, phone = ?, user_type = ?, 
                Sucursal_idSucursal = ?, email = ?, status = ? WHERE idUser = ?";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param(
            $stmt, 
            "sssiiisi", 
            $user['name'], 
            $user['username'], 
            $user['phone'], 
            $user['user_type'], 
            $user['Sucursal_idSucursal'], 
            $user['email'], 
            $user['status'], 
            $user['idUser']
        );

        $success = mysqli_stmt_execute($stmt);
        mysqli_stmt_close($stmt);
        $this->db->close($link);

        return $success;
    }

    public function getAll() { // EndPoint Creado
        $link = $this->db->open();

        $sql = "SELECT * FROM user";
        $result = mysqli_query($link, $sql);

        $users = [];
        while ($row = mysqli_fetch_assoc($result)) {
            $users[] = $row;
        }

        $this->db->close($link);
        return $users;
    }

    public function getByUserType($userType) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "SELECT * FROM user WHERE user_type = ?";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param($stmt, "i", $userType);

        mysqli_stmt_execute($stmt);
        $result = mysqli_stmt_get_result($stmt);

        $users = [];
        while ($row = mysqli_fetch_assoc($result)) {
            $users[] = $row;
        }

        mysqli_stmt_close($stmt);
        $this->db->close($link);

        return $users;
    }

    public function delete($idUser) { // EndPoint Creado
        $link = $this->db->open();

        $sql = "DELETE FROM user WHERE idUser = ?";
        $stmt = mysqli_prepare($link, $sql);
        mysqli_stmt_bind_param($stmt, "i", $idUser);

        $success = mysqli_stmt_execute($stmt);
        mysqli_stmt_close($stmt);
        $this->db->close($link);

        return $success;
    }
}

?>
